clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java

```diff
@@ -158,6 +158,26 @@ public Password getPassword(String key) {
     }
 
     /**
+     * Gets all original settings with the given prefix, leaving the prefix before adding it to the output.
+     *
+     * @param prefix the prefix to use as a filter
+     * @return a Map containing the settings with the prefix
+     * @throws ClassCastException if any of the values are not strings
+     */
+    public Map<String, String> originalsStringsWithPrefixIntact(String prefix) {
+        Map<String, String> result = new RecordingMap<>(prefix, false);
+        for (Map.Entry<String, ?> entry : originals.entrySet()) {
+            if (!(entry.getValue() instanceof String))
+                throw new ClassCastException("Non-string value found in original settings for key " + entry.getKey() +
+                        ": " + (entry.getValue() == null ? null : entry.getValue().getClass().getName()));
+
+            if (entry.getKey().startsWith(prefix) && entry.getKey().length() > prefix.length())
```

Comment:
nit: we use `entry.getKey()` several times here. Perhaps we could use a variable?